src/main/java/pl/danyboy/PasswordStage.java

```diff
@@ -2,48 +2,56 @@
 
 import java.util.ArrayList;
 import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 
 public class PasswordStage {
     private final List<Character> corectGuesses = new ArrayList<>();
 
     public String getObscuredPassword(String password, String inputFromUser) {
-        StringBuilder stringBuilder = new StringBuilder();
 
         if (inputCheck(password, inputFromUser)) return password;
 
-        addLetterToCorectGuess(password, inputFromUser);
+        addLetterToCorectGuess(inputFromUser);
 
-        createPassword(password, stringBuilder);
+        return createPassword(password, getObscuredPassword(password));
+
+    }
+
+    public String getObscuredPassword(String password) {
+        String replace = "-";
+        StringBuilder stringBuilder = new StringBuilder(password);
+        Pattern pattern = Pattern.compile("\\p{L}");
```

Comment:
Jeśli chcesz korzystać z regexpa (co jest generalnie ok) to 
```
Pattern pattern = Pattern.compile("\\p{L}");
 ```
zrób polem static final w klasie `PasswordStage`.

Kompilacje wzorca można wykonać tylko raz przy tworeniu klasy, zamiast wielokrotnie za każdym wołaniem `getObscuredPassword`. A jest to relatywnie kosztowna operacja.